api/controllers/expenseControllers.js

```diff
@@ -0,0 +1,104 @@
+const Expense = require('../models/expense');
+
+const SERVER_STATUS_ERROR = 500;
+const USER_STATUS_ERROR = 422;
+
+const expenseCreate = (req, res) => {
+  const { amount, description, budget, categories } = req.body;
+  const newExpense = new Expense({ amount, description, budget, categories });
+  newExpense
+    .save()
+    .then(expense => {
+      if (expense === null) {
+        res.status(SERVER_STATUS_ERROR).json({ error: err.message });
+      }
+      res.json(expense);
+    })
+    .catch(err => {
+      res.status(SERVER_STATUS_ERROR).json({ error: err.message });
+    });
+};
+
+const getExpense = (req, res) => {
+  const { id } = req.params;
+  Expense.findById(id)
+    .exec()
+    .then(expense => {
+      if (expense === null) {
+        res.status(USER_STATUS_ERROR).json({ error: err.message });
+      };
+      res.json(expense);
+    })
+    .catch(err => {
+      res.status(SERVER_STATUS_ERROR).json({ error: err.message });
+    });
+};
+//still workin on the next two parts 
+const budgetAdd = (req, res) => {
+  const { id } = req.params;
+  const { title, budgetAmount } = req.body;
+  const newBudget = { title, budgetAmount };
+  Expense.findById(id)
+    .exec()
+    .then(expense => {
+      if (expense === null) {
+        res.status(SERVER_STATUS_ERROR).json({ error: err.message });
+      };
+      let budgets = expense.budget
+      budgets = { newBudget }
+      expense
+        .save()
+        .then(newBudget => {
```

Comment:
I'm not sure what the second query has to do with the task. Adding a budget should be separate from adding an expense. They're different schema and models altogether. However, I believe I know exactly where you're heading with this and it starting to look great! Nice work on that.